Library/Homebrew/extend/ENV/shared.rb

```diff
@@ -196,22 +197,24 @@ def ncurses_define
 
   # @private
   def userpaths!
-    paths = self["PATH"].split(File::PATH_SEPARATOR)
-    # put Superenv.bin and opt path at the first
-    new_paths = paths.select { |p| p.start_with?("#{HOMEBREW_REPOSITORY}/Library/ENV", "#{HOMEBREW_PREFIX}/opt") }
-    # XXX hot fix to prefer brewed stuff (e.g. python) over /usr/bin.
-    new_paths << "#{HOMEBREW_PREFIX}/bin"
-    # reset of self["PATH"]
-    new_paths += paths
-    # user paths
-    new_paths += ORIGINAL_PATHS.map do |p|
-      begin
-        p.realpath.to_s
-      rescue
-        nil
-      end
-    end - %w[/usr/X11/bin /opt/X11/bin]
-    self["PATH"] = new_paths.uniq.join(File::PATH_SEPARATOR)
+    path = PATH.new(self["PATH"])
+               .select do |p|
+                 # put Superenv.bin and opt path at the first
+                 p.start_with?("#{HOMEBREW_REPOSITORY}/Library/ENV", "#{HOMEBREW_PREFIX}/opt")
+               end
+               .append(HOMEBREW_PREFIX/"bin") # XXX hot fix to prefer brewed stuff (e.g. python) over /usr/bin.
+               .append(self["PATH"]) # reset of self["PATH"]
+               .append(
+                 # user paths
+                 ORIGINAL_PATHS.map do |p|
+                   begin
+                     p.realpath.to_s
+                   rescue
+                     nil
+                   end
+                 end - %w[/usr/X11/bin /opt/X11/bin],
+               )
```

Comment:
This big long method chain is quite difficult for me to follow — is there a reason for not just writing `path.append`, etc. at the start of each line?